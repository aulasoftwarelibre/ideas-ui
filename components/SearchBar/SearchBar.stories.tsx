import { action } from '@storybook/addon-actions';
import React from 'react';

import { SearchBar } from './SearchBar';

export default {
  title: "Search Bar",
  component: SearchBar,
};

export const Default = () => (
  <SearchBar criteria="" onChange={action("clicked")}></SearchBar>
);
