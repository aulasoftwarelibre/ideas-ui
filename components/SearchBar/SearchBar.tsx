import React from "react";

export interface Props {
  criteria?: string;
  onClick?: () => void;
}

export const SearchBar: React.FunctionComponent<Props> = ({
  criteria,
  onClick,
}) => {
  return (
    <div className="flex bg-gradient-l-primary-to-light rounded-lg text-white px-2 py-2 items-center">
      <button
        id="search-tool"
        type="submit"
        className="px-3 py-3 focus:outline-none focus:shadow-outline"
        onClick={onClick}
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          className="w-6 h-6"
        >
          <path d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path>
        </svg>
      </button>
      <input
        className="bg-transparent focus:outline-none placeholder- placeholder-white text-xl w-full pl-5"
        type="search"
        name="search"
        placeholder="Buscar eventos i.e web-development"
      ></input>
    </div>
  );
};

export default SearchBar;
