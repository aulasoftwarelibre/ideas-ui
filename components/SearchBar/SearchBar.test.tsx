import { shallow } from "enzyme";
import React from "react";
import SearchBar from "./SearchBar";

describe("SearchBar", () => {
  const onClickMock = jest.fn();
  const component = shallow(<SearchBar onClick={onClickMock}></SearchBar>);

  it("should be rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should be clickable", () => {
    component.find("#search-tool").simulate("click");

    expect(onClickMock).toHaveBeenCalled();
  });
});
