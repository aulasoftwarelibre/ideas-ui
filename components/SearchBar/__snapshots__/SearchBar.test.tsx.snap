// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should be rendered correctly 1`] = `ShallowWrapper {}`;
